src/containers/offer-page/create-offer/CreateOffer.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect } from 'react'
+import { useSelector } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+
+import LeakAddSharpIcon from '@mui/icons-material/LeakAddSharp'
+import Typography from '@mui/material/Typography'
+import Box from '@mui/material/Box'
+
+import useForm from '~/hooks/use-form'
+import useConfirm from '~/hooks/use-confirm'
+import useAxios from '~/hooks/use-axios'
+import TeachingBlock from '~/containers/offer-page/create-offer/teaching-block/TeachingBlock'
+import SpecializationBlock from '~/containers/offer-page/create-offer/specialization-block/SpecializationBlock'
+import AppButton from '~/components/app-button/AppButton'
+import { OfferService } from '~/services/offer-service'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { useDrawerContext } from '~/context/drawer-context'
+
+import { snackbarVariants } from '~/constants'
+import { initialValues, validations } from '~/containers/offer-page/create-offer/CreateOffer.constants'
+import { styles } from '~/containers/offer-page/create-offer/CreateOffer.styles'
+
+export interface CreateOfferData {
+  category: string | null,
+  subject: string | null,
+  proficiencyLevel: string[],
+  languages: string[],
+  description: string,
+  price: string
+}
+
+const CreateOffer = () => {
+  const { userRole } = useSelector((state) => state.appMain)
+  const { setNeedConfirmation } = useConfirm()
+  const { closeDrawer } = useDrawerContext()
+  const { setAlert } = useSnackBarContext()
+
+  const { t } = useTranslation()
+
+  const { response, error, loading, fetchData: createOffer } = useAxios({
+    service: (offerData: CreateOfferData) => OfferService.createOffer(offerData),
+    fetchOnMount: false
+  })
+
+  const { data, errors, isDirty, handleInputChange, handleNonInputValueChange, handleBlur, handleSubmit } = useForm<CreateOfferData>(
+    { initialValues,
+      validations,
+      onSubmit: async () => {
+        await createOffer(data)
+      }
+    })
+  
+  useEffect(()=>{
+    setNeedConfirmation(isDirty)
+  },[setNeedConfirmation, isDirty])
+  
+  useEffect(() => {
+    if(error){     
+      setAlert({
+        severity: snackbarVariants.error,
+        message: `errors.${error}`
+      })
+    }
+    if(response){
+      setAlert({
+        severity: snackbarVariants.success,
+        message: 'offerPage.createOffer.successMessage'
+      })
+      closeDrawer()
+    }
+  }, [error, response, setAlert, closeDrawer])
+
+  const handleAutocompleteChange = (key: keyof CreateOfferData ) => (_: React.ChangeEvent<HTMLInputElement>, value: string | null) => {
```

Comment:
you use this type declaration in some other places, maybe it make sense to move it to independent type?

Reply:
fixed